pkg/controller/deployment/util/deployment_util_test.go

```diff
@@ -1331,3 +1331,30 @@ func TestMinAvailable(t *testing.T) {
 		})
 	}
 }
+
+func TestGetReplicaSetName(t *testing.T) {
+	hash := "6b4c6ffcfc"
+	tests := []struct {
+		name           string
+		deploymentName string
+		expected       string
+	}{
+		{
+			"deploymentName length less than 242",
+			"deploy",
+			"deploy" + "-" + hash,
+		},
+		{
+			"deploymentName length longer than 242",
+			"use-a-very-long-name-123456789-123456789-123456789-123456789-123456789-123456789-123456789-123456789-123456789-123456789-123456789-123456789-123456789-123456789-123456789-123456789-123456789-123456789-123456789-123456789-123456789-123456789-12",
+			"use-a-very-long-name-123456789-123456789-123456789-123456789-123456789-123456789-123456789-123456789-123456789-123456789-123456789-123456789-123456789-123456789-123456789-123456789-123456789-123456789-123456789-123456789-123456789-123456789-1" + "-" + hash,
+		},
+	}
```

Comment:
I'd also add testcases for input of exactly 242 (no truncation), 243 (1 tructation) and 252 (9 truncation).

Reply:
I change the testcase  code,please take a look. By the way, 252 is 10 truncation.